packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridSelector } from '../core/useGridSelector';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { GRID_NATIVE_SCROLL, GRID_ROWS_SCROLL_END } from '../../../constants/eventsConstants';
+import { gridContainerSizesSelector } from '../../root/gridContainerSizesSelector';
+import { useGridApiEventHandler } from '../../root/useGridApiEventHandler';
+import { GridRowScrollEndParams } from '../../../models/params/gridRowScrollEndParams';
+import { visibleGridColumnsSelector } from '../columns/gridColumnsSelector';
+
+export const useGridInfiniteLoader = (
+  windowRef: React.MutableRefObject<HTMLDivElement | null>,
+  apiRef: GridApiRef,
+): void => {
+  const options = useGridSelector(apiRef, optionsSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
+  const visibleColumns = useGridSelector(apiRef, visibleGridColumnsSelector);
+  const isInScrollBottomArea = React.useRef<boolean>(false);
+
+  const handleGridScroll = React.useCallback(
+    (event) => {
+      const scrollPositionBottom =
+        event.target.scrollTop + containerSizes?.windowSizes.height + options.scrollEndThreshold;
+
+      if (containerSizes && scrollPositionBottom >= containerSizes.dataContainerSizes.height) {
+        if (!isInScrollBottomArea.current) {
+          const rowScrollEndParam: GridRowScrollEndParams = {
+            api: apiRef,
+            visibleColumns,
+            viewportPageSize: containerSizes?.viewportPageSize,
+            virtualRowsCount: containerSizes?.virtualRowsCount,
+          };
+          apiRef.current.publishEvent(GRID_ROWS_SCROLL_END, rowScrollEndParam);
+          isInScrollBottomArea.current = true;
+        }
+      } else {
+        isInScrollBottomArea.current = false;
+      }
+    },
+    [options, containerSizes, apiRef, visibleColumns],
+  );
+
+  useNativeEventListener(apiRef, windowRef, GRID_NATIVE_SCROLL, handleGridScroll, {
```

Comment:
I believe that my comment and arguments were deleted

Reply:
@oliviertassinari resolved it you can check why. I don't find a problem with using the current approach. I'm not convinced with the argument of "If refactoring then -> problem" because in my mind preparing for refactoring outlines a different problem. I can spend time making the other approach work after we ship the feature but IDK how much of an effort it would be.

Reply:
I don't understand why I'm doing code reviews

Reply:
I'm not sure how to react to that comment...

Reply:
Sorry, I didn't realize that I had resolved the whole discussion. I thought that we agreed that GRID_NATIVE_SCROLL was correct and that we only needed to update the name of the event.

Reply:
either way, I'll resolve this one and discuss it under the original one.